File: Seccion 12 - Punteros en C++/Seccion_12_Dart/12_ejercicio_12.dart

import 'dart:io';

class Etapa {
  int horas, min, seg;

  Etapa(this.horas, this.min, this.seg);
}

List<Etapa> etapas = new List<Etapa>(3);

void pedirDatos() {
  int horas, minutos, segundos;

  print('Digite los tiempos empleados: ');
  for (int i = 0; i < 3; i++) {
    stdout.write('Horas: ');
    horas = int.parse(stdin.readLineSync());
    stdout.write('Minutos: ');
    minutos = int.parse(stdin.readLineSync());
    stdout.write('Segundos: ');
    segundos = int.parse(stdin.readLineSync());

    Etapa etapa = new Etapa(horas, minutos, segundos);
    etapas[i] = etapa;
    print('');
  }
}

void calcularTiempo(List<Etapa> etapas) {
  int horasT = 0, minT = 0, segT = 0;

  for (Etapa etapa in etapas) {
      horasT += etapa.horas;
      minT += etapa.min;
      if (minT >= 60) {
        minT -= 60;
        horasT++;
      }
      segT += etapa.seg;
      if (segT >= 60) {
        segT -= 60;
        minT++;
      }
  }

  print('\nMostrando el tiempo total:');
  print('Horas totales: $horasT');
  print('Minutos totales: $minT');
  print('Segundos totales: $segT');
}

void main() {

  pedirDatos();

  calcularTiempo(etapas);
  
}

File: Seccion 12 - Punteros en C++/Seccion_12_Python/12_ejercicio_12.py

class Etapa:
    horas: int
    mins: int
    segs: int
    
    def __init__(self, horas, minutos, segundos):
        self.horas = horas
        self.mins = minutos
        self.segs = segundos

etapas = []

def pedir_datos():
    global etapas
    
    print('Digite los tiempps emleados: ')
    for i in range(3):
        horas = int(input('Horas: '))
        mins = int(input('Minutos: '))
        segs = int(input('Segundos: '))
        
        etapa = Etapa(horas, mins, segs)
        etapas.append(etapa)
        print()
def calcular_tiempo(etapas: list):
    horasT: int = 0
    minT: int = 0
    segT: int = 0
    
    for etapa in etapas:
        horasT += etapa.horas
        minT += etapa.mins
        if minT >= 60:
            minT -= 60
            horasT += 1
        segT += etapa.segs
        if segT >= 60:
            segT -= 60
            minT += 1
    
    print('\nMostrando el tiempo total:')
    print('Horas: {}'.format(horasT))
    print('Minutos: {}'.format(minT))
    print('Segundos: {}'.format(segT))

def main():
    
    pedir_datos()
    
    calcular_tiempo(etapas)

if __name__ == '__main__':
    main()